cloudinit/rumcmd.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// Cmd defines a runcmd command
+type Cmd struct {
+	Cmd  string
+	Args []string
+}
+
+// UnmarshalJSON a runcmd command
+//// It can be either a list or a string. If the item is a
+//// list, it will be properly executed (with the first arg as the command).
+//// If the item is a string, it will be written to a file and interpreted using ``sh``.
+func (c *Cmd) UnmarshalJSON(data []byte) error {
+	var s1 []string
+	if err := json.Unmarshal(data, &s1); err == nil {
+		c.Cmd = s1[0]
+		c.Args = s1[1:]
+		return nil
+	}
```

Comment:
please check the err here and be sure to add a stack trace if you are returning an error from the standard library.

Reply:
In this specif case I'm intentionally ignoring the error after trying to convert to `[]string` and falling back to convert to `string` (this is necessary because Cmd can be expressed with two different formats)

Reply:
I would recommend adding code to ensure the error is exactly the expected error. This both reduced the documentation necessary because it becomes very evident what the code is doing and still allows real errors to be bubbled up from the correct location.